platform/core/src/utils/isDisplaySetReconstructable.js

```diff
@@ -70,52 +78,95 @@ function processSingleframe(instances) {
   // Check if frame spacing is approximately equal within a spacingTolerance.
   // If spacing is on a uniform grid but we are missing frames,
   // Allow reconstruction, but pass back the number of missing frames.
-  if (instances.length > 2) {
-    const lastIpp = instances[instances.length - 1].getData().metadata
+  if (n > 2) {
+    const lastIpp = instances[n - 1].getData().metadata
       .ImagePositionPatient;
 
     // We can't reconstruct if we are missing ImagePositionPatient values
-    if (!firstImagePositionPatient || !lastIpp) {
-      return { value: false };
-    }
-
-    const averageSpacingBetweenFrames =
-      _getPerpendicularDistance(firstImagePositionPatient, lastIpp) /
-      (instances.length - 1);
-
-    let previousImagePositionPatient = firstImagePositionPatient;
-
-    for (let i = 1; i < instances.length; i++) {
-      const instance = instances[i].getData().metadata;
-      const { ImagePositionPatient } = instance;
+    if (firstImagePositionPatient && lastIpp) {
+      const averageSpacingBetweenFrames =
+        _getPerpendicularDistance(firstImagePositionPatient, lastIpp) /
+        (n - 1);
+
+      let previousImagePositionPatient = firstImagePositionPatient;
+
+      for (let ii = 1; ii < n; ++ii) {
+        const instance = instances[ii].getData().metadata;
+        const { ImagePositionPatient } = instance;
+
+        const spacingBetweenFrames = _getPerpendicularDistance(
+          ImagePositionPatient,
+          previousImagePositionPatient
+        );
+        const spacingIssue = _getSpacingIssue(
+          spacingBetweenFrames,
+          averageSpacingBetweenFrames
+        );
+
+        if (spacingIssue) {
+          const issue = spacingIssue.issue;
+
+          if (issue === reconstructionIssues.MISSING_FRAMES) {
+            missingFrames += spacingIssue.missingFrames;
+          } else if (issue === reconstructionIssues.IRREGULAR_SPACING) {
+            warningIssues.push(issue);
+            break;
+          }
+        }
 
-      const spacingBetweenFrames = _getPerpendicularDistance(
-        ImagePositionPatient,
-        previousImagePositionPatient
-      );
-      const spacingIssue = _getSpacingIssue(
-        spacingBetweenFrames,
-        averageSpacingBetweenFrames
-      );
+        previousImagePositionPatient = ImagePositionPatient;
+      }
+    }
+  }
 
-      if (spacingIssue) {
-        const issue = spacingIssue.issue;
+  /*  Check if 4D dataset.
+   *
+   *  Assuming that slices at different time have the same position, here we just check if
+   *  there are multiple slices for the same ImagePositionPatient and disable MPR.
+   *
+   *  A better heuristic would be checking 4D tags, e.g. the presence of multiple TemporalPositionIdentifier values.
```

Comment:
Not needed for this pr but this part of the code looks almost like a jsDOC for a function, when possible, create documented helper functions like _check4DDataset or something. This helps reusability, readability and also will hide complex logic :)

Reply:
done